clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -270,6 +275,27 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
         return getKeySerializerWithRetryWithShortInterval();
       } catch (Exception e) {
         whetherStoreInitTriggeredByRequestFail = true;
+        // Kick off an async thread to keep retrying
+        asyncStoreInitThread = new Thread(() -> {
+          while (true) {
+            try {
+              getKeySerializerWithRetryWithShortInterval();
+              whetherStoreInitTriggeredByRequestFail = false;
+              LOGGER.info("Successfully init store client by async store init thread");
```

Comment:
Wouldn't this always trigger even if the initialization was successful?

Reply:
It shouldn't.
If the first store init succeeds, `keySerializer` won't be `null`, so this function will return directly and the async store init won't be triggered.

The async store init will only be triggered if the store init triggered by start and the first request fails.